src/index.ts

```diff
@@ -94,22 +94,33 @@ Cypress.Commands.add(
             "Currently only support fetch(url, options), saw fetch(Request)"
           );
         }
+
         if (input.indexOf(endpoint) !== -1 && init && init.method === "POST") {
-          const payload: GQLRequestPayload<AllOperations> = JSON.parse(
+          let json = JSON.parse(
             init.body as string
           );
-          const { operationName, query, variables } = payload;
-          return graphql({
-            schema,
-            source: query,
-            variableValues: variables,
-            operationName,
-            rootValue: getRootValue<AllOperations>(
-              currentOps,
+          json = Array.isArray(json) ? json : [json];
+          return Promise.all(json.map((payload: GQLRequestPayload<AllOperations>) => {
+            const { operationName, query, variables } = payload;
+            return graphql({
+              schema,
+              source: query,
+              variableValues: variables,
               operationName,
-              variables
-            )
-          }).then((data: any) => new Response(JSON.stringify(data)));
+              rootValue: getRootValue<AllOperations>(
+                currentOps,
+                operationName,
+                variables
+              )
+            });
+          })).then(dataArray => new Response(JSON.stringify(
+            dataArray.reduce((prev: any, data: any) => ({
+              data: {
+                ...prev.data,
+                ...data.data,
+              },
```

Comment:
Is this the correct behavior of batch? It just merges the response data together? What if there are overlapping fields?

Reply:
I don't think the reduce is necessary... something like this worked for me:

```
return Promise.all().then(results => {
  return responseStub(results);
});

function responseStub(result) {
  return Promise.resolve({
    json() {
      return Promise.resolve(result);
    },
    text() {
      return Promise.resolve(JSON.stringify(result));
    },
    ok: true,
  });
}
```

I'm guessing the responseStub is not necessary, but it seems common.

Reply:
It does not merge the response fields but rather the queries into one object, you can't really send 2 queries with the same name I guess.

But I don't know about the exact behavior of batch to be honest, I checked the response format I'm getting from my server and changed this PR accordingly :)

Reply:
I think it may depend on the GraphQL server. Newer releases of Apollo's GraphQL server [accept and return an array of queries](https://blog.apollographql.com/query-batching-in-apollo-63acfd859862), rather than merging them, to avoid the issue of field collision. I've implemented support for that method of query batching in #10.